curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorFrameworkImpl.java

```diff
@@ -200,23 +232,39 @@ public QuorumVerifier getCurrentConfig()
         return (ensembleTracker != null) ? ensembleTracker.getCurrentConfig() : null;
     }
 
-    private ZookeeperFactory makeZookeeperFactory(final ZookeeperFactory actualZookeeperFactory)
+    private ZookeeperFactory makeZookeeperFactory(final ZookeeperFactory actualZookeeperFactory, ZKClientConfig zkClientConfig)
     {
+    	if (actualZookeeperFactory instanceof ConfigurableZookeeperFactory) {
```

Comment:
I'm -1 on using `instanceof`. Wouldn't it be better to add a `default` method to `ZookeeperFactory` that takes the `ZKClientConfig`? It could be optional and forward to the older versions when not present. The default version in `DefaultZookeeperFactory` would throw an exception.

Reply:
Understood. Done.